packages/dashboard/src/components/SubwayNav/SubwayNav.tsx

```diff
@@ -0,0 +1,298 @@
+import * as React from 'react';
+
+import { classNamesFunction, css } from 'office-ui-fabric-react/lib/Utilities';
+import { ISubwayNavProps, ISubwayNavStep, ISubwayNavStyles, SubwayNavStepState } from './SubwayNav.types';
+import { Icon, Link } from 'office-ui-fabric-react';
+import { getSubwayNavStyles } from './SubwayNav.styles';
+
+export interface IIconProps {
+  iconName: string;
+
+  iconClassName: string;
+}
+
+const getClassNames = classNamesFunction<ISubwayNavProps, ISubwayNavStyles>();
+const classNames = getClassNames(getSubwayNavStyles!);
+
+/**
+ * Component for Subway control
+ */
+export class SubwayNav extends React.Component<ISubwayNavProps, {}> {
+  constructor(props: ISubwayNavProps) {
+    super(props);
+  }
+
+  /**
+   * Render the Subway Nav control
+   */
+  public render(): JSX.Element {
+    const { steps /*theme, */ } = this.props;
+    let stepElements: JSX.Element[] = [];
+    let renderedSteps: JSX.Element[] = [];
+
+    if (steps) {
+      for (let stepIndex = 0; stepIndex < steps.length; stepIndex++) {
+        renderedSteps = this._renderStep(steps[stepIndex], stepIndex === 0 ? undefined : steps[stepIndex - 1]);
+        stepElements = stepElements.concat(renderedSteps);
+      }
+
+      return (
+        <div className={css(this.props.className, classNames.subwayNavContainer)}>
+          <div className={classNames.subwayNavContentContainer}>
+            <div className={classNames.subwayNavContent}>{stepElements}</div>
+          </div>
+        </div>
+      );
+    } else {
+      // Empty element
+      return <React.Fragment />;
+    }
+  }
+
+  /**
+   * Render step of the subway Nav control
+   * @param stepToRender
+   * @param prevStep
+   */
+  private _renderStep(stepToRender: ISubwayNavStep, prevStep: ISubwayNavStep | undefined): JSX.Element[] {
+    let stepElements: JSX.Element[] = [];
+
+    if (stepToRender === undefined) {
+      return stepElements;
+    }
+
+    const { steps } = this.props;
+
+    if (steps !== undefined) {
+      const iconProps = this._getIconPropsStep(stepToRender);
+
+      if (prevStep !== undefined) {
+        const connectorClass = this._getStepConnectorClassName(stepToRender);
+        stepElements.push(
+          <div className={css(classNames.subwayNavStepConnector, connectorClass)} key={'connector' + '-' + stepToRender.key} />
```

Comment:
A pseudo element seems better suited for this purpose.
@brandonthomas please kindly comment.

Reply:
I'll likely be doing this as a single element positioned absolutely due to the fact that the animations need to be eased and continuous. Coordinating that across many items requires a lot of calculations instead of just using a CSS animation with an easing function. We could do the single line as a pseudo though.